rskj-core/src/main/java/org/ethereum/util/TransactionArgumentsUtil.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.ethereum.util;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.ethereum.core.Account;
+import org.ethereum.core.TransactionArguments;
+import org.ethereum.core.TransactionPool;
+import org.ethereum.rpc.TypeConverter;
+import org.ethereum.rpc.Web3;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+import org.ethereum.vm.GasCost;
+
+public class TransactionArgumentsUtil {
+
+	private static final BigInteger DEFAULT_GAS_LIMIT = BigInteger.valueOf(GasCost.TRANSACTION_DEFAULT);
+
+	public static final String ERR_INVALID_CHAIN_ID = "Invalid chainId: ";
+
+	/**
+	 * transform the Web3.CallArguments in TransactionArguments that can be used in
+	 * the TransactionBuilder
+	 */
+	public static TransactionArguments processArguments(Web3.CallArguments argsParam, TransactionPool transactionPool, Account senderAccount, byte defaultChainId) {
+
+		TransactionArguments argsRet = new TransactionArguments();
+
+		argsRet.setFrom(argsParam.from);
+
+		argsRet.setTo(stringHexToByteArray(argsParam.to));
+
+		argsRet.setNonce(stringNumberAsBigInt(argsParam.nonce, () -> transactionPool.getPendingState().getNonce(senderAccount.getAddress())));
+
+		argsRet.setValue(stringNumberAsBigInt(argsParam.value, () -> BigInteger.ZERO));
+
+		argsRet.setGasPrice(stringNumberAsBigInt(argsParam.gasPrice, () -> BigInteger.ZERO));
+
+		argsRet.setGasLimit(stringNumberAsBigInt(argsParam.gas, () -> null));
+
+		if (argsRet.getGasLimit() == null) {
+			argsRet.setGasLimit(stringNumberAsBigInt(argsParam.gasLimit, () -> DEFAULT_GAS_LIMIT));
+		}
+
+		if (argsParam.data != null && argsParam.data.startsWith("0x")) {
+			argsRet.setData(argsParam.data.substring(2));
+			argsParam.data = argsRet.getData(); // needs to change the parameter because some places expect the changed value after sendTransaction call
+		}
+
+		argsRet.setChainId(hexToChainId(argsParam.chainId));
+		if (argsRet.getChainId() == 0) {
+			argsRet.setChainId(defaultChainId);
+		}
+
+		return argsRet;
+	}
+
+	private static BigInteger stringNumberAsBigInt(String number, Supplier<BigInteger> getDefaultValue) {
+
+		BigInteger ret = Optional.ofNullable(number).map(TypeConverter::stringNumberAsBigInt).orElseGet(getDefaultValue);
+
+		return ret;
+	}
+
+	private static byte[] stringHexToByteArray(String value) {
+
+		byte[] ret = Optional.ofNullable(value).map(TypeConverter::stringHexToByteArray).orElse(null);
+
+		return ret;
+	}
+
+	private static byte hexToChainId(String hex) {
+		if (hex == null) {
+			return 0;
+		}
+		try {
+			byte[] bytes = TypeConverter.stringHexToByteArray(hex);
+			if (bytes.length != 1) {
+				throw RskJsonRpcRequestException.invalidParamError(ERR_INVALID_CHAIN_ID + hex);
+			}
+
+			return bytes[0];
+		} catch (Exception e) {
```

Comment:
why would we catch exceptions?
I mean if it's an unknown exception, I think we should let it explode upwards
and If it's a RskJsonRpcRequestException, we'll wrap it with another RskJsonRpcRequestException on top (an exception with another exception of the same type nested inside) - we don't want this right?

Reply:
This part of the code was introduced while i was validanting my changes and i just merged it. But i see your point